internal/auditlog/serial_writer.go

```diff
@@ -0,0 +1,52 @@
+// Copyright The OWASP Coraza contributors
+// SPDX-License-Identifier: Apache-2.0
+
+package auditlog
+
+import (
+	"io"
+
+	"github.com/corazawaf/coraza/v3/experimental/plugins"
+	"github.com/corazawaf/coraza/v3/experimental/plugins/plugintypes"
+	"github.com/tetratelabs/proxy-wasm-go-sdk/proxywasm"
+)
+
+// Coraza does not come with a built-in audit log writer for Wasm
+// See https://github.com/corazawaf/coraza/blob/main/internal/auditlog/init_tinygo.go
+// This function registers a new audit log writer for Wasm named "wasmserial" that prints
+// audit logs to the proxy-wasm log as info messages.
+func RegisterWasmSerialWriter() {
```

Comment:
Just wondering, what's the difference in the output if not using the custom writer, i.e. using stdout or err instead of `proxywasm.LogInfo`? I thought the latter might pack everything onto a single line but it seems it doesn't, so don't know if there's a big difference.

Reply:
Thanks Rag, it is actually working fine, I added to the Coraza PR the registration of the Serial writer also for the tinygo build: https://github.com/corazawaf/coraza/pull/1027#discussion_r1543367048
I see value in overriding it only if we wish to add some prefixes, or something to the line to highlight that it is an audit log (already considering that we are mixing all the logs into the normal envoy logs as a start.

Reply:
Sounds good, I think a prefix makes sense